docs/Roadmap.md

```diff
@@ -0,0 +1,122 @@
+        WASI-compatible? What are good incremental steps?
+    *   Are there any performance gaps?
+*   (Help wanted) Evaluate uses of foreign functions to identify feature gaps.
+    *   For example, Envoy
+        [registers foreign functions](https://github.com/search?q=repo%3Aenvoyproxy%2Fenvoy%20RegisterForeignFunction&type=code)
+        for signature checking, compression, filter state, route cache, and CEL
+        expressions.
+    *   Are there similar extensions in Nginx? Apache Traffic Server?
+    *   Which of these features should be promoted to ABI interfaces?
+
+<a name="sdks"></a>
+
+## SDKs / language support
+
+*   (Q1'25: @leonm1) Fork the abandoned Go SDK + support full Golang.
+*   (Google exploring) Build a Python SDK using a MicroPython port.
+*   (Help wanted) Stop using Emscripten in the C++ SDK. Instead use Clang /
+    wasi-sdk (proxy-wasm/proxy-wasm-cpp-sdk#167).
+*   (Help wanted) Merge LeakSignal's
+    [proxy-sdk](https://crates.io/crates/proxy-sdk) crate into the Rust SDK.
+*   (Help wanted) Build a Lua SDK using a Lua interpreter.
+    *   Seems quite feasible given projects like
+        [wasm_lua](https://github.com/vvanders/wasm_lua).
+    *   Could replace Envoy's
+        [Lua filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter)
+    *   Could benefit NGINX's Lua-based [OpenResty](https://openresty.org/)
+        ecosystem
+*   (Help wanted) Optimize Rust SDK binary size. It seems compiler dead-code
+    elimination is thwarted by the use of Context traits.
+
+<a name="host"></a>
+
+## Host features
+
+*   (Q1'25: @mpwarres) CppHost maintenance.
+    *   Update v8 and upstream some patches for v8 warming / extension.
+    *   Update the protobuf dependency.
+    *   Set up dependabot.
+*   (Help wanted) Prototype
+    [HyperLight](https://github.com/hyperlight-dev/hyperlight) as a KVM-based
+    sandboxing layer around wasm runtimes. The allure is getting an inexpensive
+    and transparent second layer of security at a thread boundary, which makes
+    it more feasible to run fully untrusted workloads with Proxy-Wasm.
+*   (Help wanted) Performance benchmarks. One of Proxy-Wasm's strengths is its
+    ability to swap between multiple wasm runtimes. Help users make an informed
+    decision by benchmarking cold start and execution costs across runtimes.
+*   (Help wanted) Adopt CPU metering as a first-class feature. Leverage
+    instruction counting where available. For other engines (e.g. v8), use a
+    watchdog thread.
+*   (Help wanted) Support dynamic (per VM) limits for RAM and CPU.
+*   (Help wanted) Expand the use of SharedArrayBuffer to reduce memcpy into wasm
+    runtimes. This is especially promising for HTTP body chunks. See relevant
+    [WASI issue](https://github.com/WebAssembly/WASI/issues/594). Also reduce
+    [binary memcpy](https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/21a5b089f136712f74bfa03cde43ae8d82e066b6/src/v8/v8.cc#L272).
```

Comment:
This is slightly more complicated than it initially appears... Supporting this and eliminating copies would require host (including it's modules/extensions) to read/write data from/to the network to/from memory shared with WasmVM, which is linear memory block that needs to be pre-allocated upfront.